### PR TITLE
feat: api add challenge flavor

### PR DESCRIPTION
**Pull request by Z-a-r-a-k-i**

### CODE REVIEW

File: go/cmd/pathwar/main.go [hunk 1]
```diff
@@ -648,6 +663,84 @@ func main() {
...
+					return err
+				}
+
+				ctx := context.Background()
+				apiClient, err := httpClientFromEnv(ctx)
+				if err != nil {
+					return errcode.TODO.Wrap(err)
+				}
+
+				challengeID, err := strconv.ParseInt(addChallengeFlavorChallengeID, 10, 64)
```
Comment (moul) [inline go/cmd/pathwar/main.go]: you can avoid parsing by using an int flag

File: go/pkg/pwapi/api_admin-challenge-instance-add.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	if err != nil {
+		return nil, errcode.ErrChallengeInstanceAdd.Wrap(err)
+	}
+
+	challengeInstance := pwdb.ChallengeInstance{
+		Status:   in.ChallengeInstance.Status,
+		Agent:    &agent,
+		AgentID:  in.ChallengeInstance.AgentID,
+		Flavor:   &flavor,
+		FlavorID: in.ChallengeInstance.FlavorID,
```
Comment (moul) [inline go/pkg/pwapi/api_admin-challenge-instance-add.go]:
```suggestion
		Status:   in.ChallengeInstance.Status,
		AgentID:  in.ChallengeInstance.AgentID,
		FlavorID: in.ChallengeInstance.FlavorID,
```

use only the ID here

File: go/pkg/pwapi/api_admin-challenge-flavor-add.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		IsDraft:          in.ChallengeFlavor.IsDraft,
+		IsLatest:         in.ChallengeFlavor.IsLatest,
+		SourceURL:        in.ChallengeFlavor.SourceURL,
+		ComposeBundle:    in.ChallengeFlavor.ComposeBundle,
+		Driver:           in.ChallengeFlavor.Driver,
+		Challenge:        in.ChallengeFlavor.Challenge,
+		ChallengeID:      in.ChallengeFlavor.ChallengeID,
+		SeasonChallenges: in.ChallengeFlavor.SeasonChallenges,
+	}
+
```
Comment (moul) [inline go/pkg/pwapi/api_admin-challenge-flavor-add.go]:
```suggestion
	challengeFlavor := pwdb.ChallengeFlavor{
		Version:          in.ChallengeFlavor.Version,
		Changelog:        in.ChallengeFlavor.Changelog,
		IsDraft:          in.ChallengeFlavor.IsDraft,
		IsLatest:         in.ChallengeFlavor.IsLatest,
		SourceURL:        in.ChallengeFlavor.SourceURL,
		ComposeBundle:    in.ChallengeFlavor.ComposeBundle,
		Driver:           in.ChallengeFlavor.Driver,
		Challenge:        in.ChallengeFlavor.Challenge,
		ChallengeID:      in.ChallengeFlavor.ChallengeID,
		SeasonChallenges: in.ChallengeFlavor.SeasonChallenges,
	}
    challengeFlavor := in.ChallengeFlavor
```

File: go/pkg/pwapi/api_admin-challenge-instance-add.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
```
Comment (moul) [inline go/pkg/pwapi/api_admin-challenge-instance-add.go]:
```suggestion
    challengeInstance := in.ChallengeInstance
```

File: go/pkg/pwapi/api_admin-challenge-flavor-add.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+func (svc *service) AdminChallengeFlavorAdd(ctx context.Context, in *AdminChallengeFlavorAdd_Input) (*AdminChallengeFlavorAdd_Output, error) {
+	if !isAdminContext(ctx) {
+		return nil, errcode.ErrRestrictedArea
+	}
+
+	if in == nil {
+		return nil, errcode.ErrMissingInput
+	}
+
```
Comment (moul) [inline go/pkg/pwapi/api_admin-challenge-flavor-add.go]: check if ChallengeID is set and exists, else, return an error

File: go/cmd/pathwar/main.go (hunk 1 above)
Comment (Z-a-r-a-k-i) [reply go/cmd/pathwar/main.go]: Thanks !

File: go/pkg/pwapi/api_admin-challenge-flavor-add.go (hunk 3 above)
Comment (Z-a-r-a-k-i) [reply go/pkg/pwapi/api_admin-challenge-flavor-add.go]: Thanks

File: go/pkg/pwapi/api_admin-challenge-instance-add.go [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+		Preload("Agent").
+		First(&agent).
+		Error
+	if err != nil {
+		return nil, errcode.ErrChallengeInstanceAdd.Wrap(err)
+	}
+
+	var flavor pwdb.ChallengeFlavor
+	err = svc.db.
+		Where(pwdb.SeasonChallenge{ID: in.ChallengeInstance.FlavorID}).
```
Comment (moul) [inline go/pkg/pwapi/api_admin-challenge-instance-add.go]:
```suggestion
		Where(pwdb.ChallengeFlavor{ID: in.ChallengeInstance.FlavorID}).
```

File: go/pkg/pwapi/api_admin-challenge-instance-add.go [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+		return nil, errcode.ErrRestrictedArea
+	}
+
+	if in == nil {
+		return nil, errcode.ErrMissingInput
+	}
+
+	var agent pwdb.Agent
+	err := svc.db.
+		Where(pwdb.SeasonChallenge{ID: in.ChallengeInstance.AgentID}).
```
Comment (moul) [inline go/pkg/pwapi/api_admin-challenge-instance-add.go]:
```suggestion
		Where(pwdb.Agent{ID: in.ChallengeInstance.AgentID}).
```

File: go/pkg/pwapi/api_admin-challenge-instance-add.go [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+		First(&agent).
+		Error
+	if err != nil {
+		return nil, errcode.ErrChallengeInstanceAdd.Wrap(err)
+	}
+
+	var flavor pwdb.ChallengeFlavor
+	err = svc.db.
+		Where(pwdb.SeasonChallenge{ID: in.ChallengeInstance.FlavorID}).
+		Preload("ChallengeFlavor").
```
Comment (moul) [inline go/pkg/pwapi/api_admin-challenge-instance-add.go]:
```suggestion
```

File: go/pkg/pwapi/api_admin-challenge-instance-add.go [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+	}
+
+	if in == nil {
+		return nil, errcode.ErrMissingInput
+	}
+
+	var agent pwdb.Agent
+	err := svc.db.
+		Where(pwdb.SeasonChallenge{ID: in.ChallengeInstance.AgentID}).
+		Preload("Agent").
```
Comment (moul) [inline go/pkg/pwapi/api_admin-challenge-instance-add.go]:
```suggestion
```

File: go/pkg/pwapi/api_admin-challenge-flavor-add.go [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+	var challenge pwdb.Challenge
+	err := svc.db.
+		Where(pwdb.Challenge{ID: in.ChallengeInstance.ChallengeID}).
```
Comment (moul) [inline go/pkg/pwapi/api_admin-challenge-flavor-add.go]:
```suggestion
		Where(pwdb.Challenge{ID: in.ChallengeFlavor.ChallengeID}).
```

**Pull request merged**